pkgs/checks/lib/src/checks.dart

```diff
@@ -187,40 +187,204 @@ extension ContextExtension<T> on Subject<T> {
   Context<T> get context => _context;
 }
 
-/// The expectation and nesting context already applied to a [Subject].
+/// The context for a [Subject] that allows asserting expectations and creating
+/// nested subjects.
 ///
-/// This is the surface of interaction for expectation extension method
-/// implementations.
+/// A [Subject] is the target for checking expectations in a test.
+/// Every subject has a [Context] which holds the "actual" value, tracks how the
+/// value was obtained, and can check expectations about the value.
 ///
-/// The `expect` and `expectAsync` can test the value and optionally reject it.
-/// The `nest` and `nestAsync` can test the value, and also extract some other
-/// property from it for further checking.
+/// The user focused APIs called within tests are expectation extension methods
+/// written in extension `on Subject`, typically specialized to a specific
+/// generic.
+///
+/// Expectation extension methods will make a call to one of the APIs on the
+/// subject's [Context], and can perform one of two types of operations:
+///
+/// -   Expect something of the current value.
+/// -   Expect that a new subject can be extracted from the current value.
+///
+///
+/// Whichever type of operation, an expectation extension method provides two
+/// callbacks.
```

Comment:
I think some sort of link to the api  here would be good, to show how these are provided, it's a bit confusing without that imo. I end up wanting to interrupt by reading to find that.

Reply:
I expanded this to show one callback signature, and describe that the other signature varies. Does that help?

Reply:
What I am confused by here isn't what the callbacks are supposed to do, its how/where they are provided.

I think this stems from the mentions of `expect` etc being below this - these are describing how those work right? Feels like its putting the cart before the horse a bit, I don't have the context yet for what is being described.

Reply:
How about mentioning each method in the bullet list above of the two types of operations?